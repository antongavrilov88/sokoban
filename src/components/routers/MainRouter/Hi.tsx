import React, { FC } from 'react'

export const Hi: FC = () => {
    
    return (
        <div >
            Hi
        </div>  
    )
}
