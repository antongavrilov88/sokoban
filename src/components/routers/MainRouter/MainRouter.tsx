import React, { FC } from 'react'
import { Route, Switch } from 'react-router-dom'
import { PageError, PageForum, PageGame, PageHome, PageLeaderboard, PageProfile, PageSignin, PageSignup, 
    routeForum, routeGame, routeHome, routeLeaderboard, routeProfile, routeSignin, routeSignup } from './constants'

export const MainRouter: FC = () => {
    
    return (
        <Switch>
            <Route path={ routeHome }         exact component={ PageHome } />
            <Route path={ routeLeaderboard }  exact component={ PageLeaderboard } />
            <Route path={ routeForum }        exact component={ PageForum } />
            <Route path={ routeGame }         exact component={ PageGame } />
            <Route path={ routeProfile }      exact component={ PageProfile } />
            <Route path={ routeSignup }       exact component={ PageSignup } />
            <Route path={ routeSignin }       exact component={ PageSignin } />
            <Route component={ PageError } />
        </Switch>
    )
}
