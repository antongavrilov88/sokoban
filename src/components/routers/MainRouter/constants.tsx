import React from 'react'
import {Leaderboard} from '../../../pages/Leaderboard/index'
import {SignInPage} from '../../../pages/SignIn/SignIn'
import {LayOut} from '../../LayOut/index'
import {SignUpPage} from '../../../pages/SignUp/SignUp'
import { Profile } from '../../../pages/Profile/index'
import { Err } from '../../../pages/Err/index'

export const PageHome           = () => <h1>home</h1>
export const PageLeaderboard    = Leaderboard
export const PageForum          = () => <h1>forum</h1>
export const PageGame           = LayOut
export const PageProfile        = Profile
export const PageSignup         = SignUpPage
export const PageSignin         = SignInPage
export const PageError          = Err

export const routeHome          = '/'
export const routeLeaderboard   = '/leaderboard'
export const routeForum         = '/forum'
export const routeGame          = '/game'
export const routeProfile       = '/profile'
export const routeSignup        = '/signup'
export const routeSignin        = '/signin'
