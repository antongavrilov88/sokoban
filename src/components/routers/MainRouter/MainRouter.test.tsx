import { MemoryRouter, Route } from 'react-router-dom'
import React, { ReactElement } from 'react'
import { MainRouter } from './MainRouter'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

Enzyme.configure({ adapter: new Adapter() })

describe.skip('MainRouter: snapshot', () => {
    
    const component = mount(<MainRouter />)
    expect(component).toMatchSnapshot()

})



describe.skip('MainRouter: wrapper', () => {
    
    const getWrapper = (route: string): ReactElement => {
        return <MemoryRouter initialEntries={[route]}>
                    <MainRouter />
                </MemoryRouter>
    }
    
    test('should render the home page', () => {
        const page = <h1>/home</h1>
        expect(mount(getWrapper('/')).contains(page)).toEqual(true)
    })

    // Всегда рендерит /. Не понятно.
    test('should render the leaderboard page', () => {
        const page = <h1>/leaderboard</h1>
        // Ошибка 
        expect(mount(getWrapper('/leaderboard')).contains(page)).toEqual(true)
        // Нет ошибки
        expect(mount(getWrapper('/leaderboard')).contains(<h1>/home</h1>)).toEqual(true)
        
        //expect(mount(getWrapper('/leaderboard'))).toMatchSnapshot()
    })

})

let pathMap: any = {}
describe.skip('MainRouter: array of routes', () => {
    
    beforeAll(() => {
        
        const component = shallow(<MainRouter/>);
        pathMap = component.find(Route).reduce((pathMap, currRoute) => {
            const routeProps: any = currRoute.props()
            pathMap[routeProps.path] = routeProps.component;
            return pathMap;
        }, pathMap)
        
    })

    test('should render the home page', () => {
        const page = function() { return <h1>/home</h1> } 
        expect((pathMap['/'])).toBe(page)
    })
    
})
