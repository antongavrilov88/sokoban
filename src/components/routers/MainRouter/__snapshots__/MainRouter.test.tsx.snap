// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/leaderboard"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/forum"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/game"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/profile"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/auth"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
